x-pack/plugins/alerting/server/alerts_service/lib/set_alerts_to_untracked.ts

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import { Logger } from '@kbn/logging';
+import {
+  ALERT_RULE_UUID,
+  ALERT_STATUS,
+  ALERT_STATUS_ACTIVE,
+  ALERT_STATUS_UNTRACKED,
+  ALERT_UUID,
+} from '@kbn/rule-data-utils';
+
+export interface SetAlertsToUntrackedOpts {
+  indices: string[];
+  ruleIds?: string[];
+  alertUuids?: string[];
+}
+
+type UntrackedAlertsResult = Array<{ [ALERT_RULE_UUID]: string; [ALERT_UUID]: string }>;
+
+export async function setAlertsToUntracked({
+  logger,
+  esClient,
+  indices,
+  ruleIds = [],
+  alertUuids = [], // OPTIONAL - If no alertUuids are passed, untrack ALL ids by default
+}: {
+  logger: Logger;
+  esClient: ElasticsearchClient;
+} & SetAlertsToUntrackedOpts): Promise<UntrackedAlertsResult> {
+  if (!ruleIds && !alertUuids) throw new Error('Must provide either ruleIds or alertUuids');
```

Comment:
Would it make sense to check that one or the either is defined and non-empty? Are there use cases where we're specifying a rule ID and an alert UUID?

Reply:
No there should be no use cases where we're passing both.